pennylane_lightning/core/src/simulators/lightning_tensor/tncuda/gates/TNCudaGateCache.hpp

```diff
@@ -108,8 +113,29 @@ template <class PrecisionT> class TNCudaGateCache {
             std::piecewise_construct, std::forward_as_tuple(gate_id),
             std::forward_as_tuple(gate_key, std::move(tensor)));
 
-        device_gates_.at(gate_id).second.getDataBuffer().CopyHostDataToGpu(
-            gate_data_host.data(), gate_data_host.size());
+        if (adjoint) {
+            // TODO: This is a temporary solution for gates data transpose.
+            // There should be a better way to handle this, but there is not
+            // a big performance issue for now since the size of gates is small.
+            std::vector<CFP_t> data_host_transpose(gate_data_host.size());
+
+            std::size_t col_size = 1 << (rank / 2);
+            std::size_t row_size = 1 << (rank / 2);
```

Comment:
Since it is "square", could we do
```suggestion
            const std::size_t n = 1 << (rank / 2);
```

Reply:
Sure!